crates/miden-lib/src/transaction/inputs.rs

```diff
@@ -229,6 +241,19 @@ impl TransactionAdviceInputs {
         let code = account.code();
         self.add_map_entry(code.commitment(), code.as_elements());
 
+        // Insert the account advice map into the advice recorder.
+        // This ensures that the advice map is available during the note script execution when it
+        // calls the account's code that relies on the it's advice map data (data segments) loaded
+        // into the advice provider
+        self.extend_map(
+            account
+                .code()
+                .mast()
+                .advice_map()
+                .iter()
+                .map(|(key, values)| (*key, values.clone())),
+        );
```

Comment:
Could we note use something like this here?:
```rust
self.0.map.merge_advice_map(account.code().mast().advice_map())
```
I think a complication may be that this returns a result - but this may be important to handle. If we merge two advice maps that contain conflicting data, the programs may not execute correctly and we won't know why. So, we should probably handle these errors properly (either here or in a different place).

Reply:
> (either here or in a different place)

I think doing this somewhere else would defeat the purpose of having a `TransactionAdviceInputs`, unless I'm thinking about this incorrectly and these inputs are somewhat optional or something like that.